webserver/__init__.py

```diff
@@ -131,6 +136,20 @@ def before_request_gdpr_check():
         elif current_user.is_authenticated and current_user.gdpr_agreed is None:
             return redirect(url_for('index.gdpr_notice', next=request.full_path))
 
+    @app.before_request
+    def prod_https_login_redirect():
+        """ Redirect to HTTPS in production except for the API endpoints
+        """
+        split_url = urlparse.urlsplit(request.url)
+        if request.headers.get('X-Forwarded-Proto') == 'http' and not request.path.startswith(API_PREFIX):
```

Comment:
I'd like to do this only in production, why can't we only do it if DEBUG is false?
This is going to cause problems if we do it on POST operations, this should only redirect on GET.
The important API endpoint that we don't want to redirect on is the lowlevel submission endpoint: https://github.com/metabrainz/acousticbrainz-server/blob/46f670d62546ab44aa9086c2724aba57d41e5387/webserver/views/api/legacy.py#L54 which is not under the API_PREFIX. There are other GET methods here that we shouldn't redirect on either.
If we only redirect when DEBUG is false, why can't you get the url directly from the request url instead of special-casing the main and beta domains?